core/templates/services/user-backend-api.service.spec.ts

```diff
@@ -0,0 +1,294 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests that the user service is working as expected.
+ */
+
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { CsrfTokenService } from 'services/csrf-token.service';
+
+import { UserBackendApiService } from 'services/user-backend-api.service';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { UserInfoObjectFactory } from 'domain/user/UserInfoObjectFactory';
+import { UrlService } from './contextual/url.service';
```

Comment:
import order:

1. Angular imports
2. Third-party imports
3. Application imports.

Within these sub-categories: sort lexicographically based on module (i.e. where it is imported from, not what is being imported).
For example:
#### Right way (because 'B' < 'D')
```
import { C } from 'B';
import { A } from 'D'; 
```

#### Wrong way
```
import { A } from 'D';
import { C } from 'B'; 
```

```suggestion
import { HttpClientTestingModule, HttpTestingController } from
  '@angular/common/http/testing';
import { TestBed, fakeAsync, flushMicrotasks } from '@angular/core/testing';

import { UrlInterpolationService } from
  'domain/utilities/url-interpolation.service';
import { UserInfoObjectFactory } from 'domain/user/UserInfoObjectFactory';
import { WindowRef } from 'services/contextual/window-ref.service';
import { CsrfTokenService } from 'services/csrf-token.service';
import { UserBackendApiService } from 'services/user-backend-api.service';
import { UrlService } from './contextual/url.service';
```